FinniversKit/Sources/Components/FrontPageSavedSearchesView/FrontPageSavedSearchesView.swift

```diff
@@ -0,0 +1,183 @@
+
+    // MARK: - Internal properties
+
+    static let topPadding: CGFloat = .spacingL
+    static let height: CGFloat = headerHeight + cellHeight + bottomPadding
+
+    // MARK: - Private properties
+
+    private static let headerHeight: CGFloat = 44
+    private static let cellHeight: CGFloat = 100
+    private static let bottomPadding: CGFloat = .spacingS
+
+    private typealias Datasource = UICollectionViewDiffableDataSource<Section, AnyHashable>
+    private typealias Snapshot = NSDiffableDataSourceSnapshot<Section, AnyHashable>
+
+    private enum Section: Int, CaseIterable {
+        case savedSearch
+    }
+
+    private var collectionViewDatasource: Datasource!
+    private weak var remoteImageDataSource: RemoteImageViewDataSource?
+    private var scrollToIndexPath: IndexPath?
+    private let title: String
+    private let buttonTitle: String
+
+    private var compositionalLayout: UICollectionViewCompositionalLayout {
+        let layout = UICollectionViewCompositionalLayout { [weak self] _, _ in
+            return self?.horizontalLayoutSection
+        }
+
+        let config = UICollectionViewCompositionalLayoutConfiguration()
+        layout.configuration = config
+        return layout
+    }
+
+    private lazy var collectionView: UICollectionView = {
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: compositionalLayout)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.delegate = self
+        collectionView.register(FrontPageSavedSearchCell.self)
+        collectionView.register(FrontPageHeaderView.self, ofKind: FrontPageHeaderView.reuseIdentifier)
+        return collectionView
+    }()
+
+    // MARK: - Init
+
+    public init(
+        title: String,
+        buttonTitle: String,
+        remoteImageDataSource: RemoteImageViewDataSource?,
+        withAutoLayout: Bool = false
+    ) {
+        self.title = title
+        self.buttonTitle = buttonTitle
+        self.remoteImageDataSource = remoteImageDataSource
+        super.init(frame: .zero)
+        translatesAutoresizingMaskIntoConstraints = !withAutoLayout
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Lifecycle
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+        if let indexPath = scrollToIndexPath {
+            self.scrollToIndexPath = nil
+            // Using centeredHorizontally for the first cell to preserve leading content inset for iOS 14 and below.
+            let scrollPosition: UICollectionView.ScrollPosition = indexPath.item == 0 ? .centeredHorizontally : .left
+            collectionView.scrollToItem(at: indexPath, at: scrollPosition, animated: false)
+        }
+    }
+}
+
+// MARK: - Public methods
+
+public extension FrontPageSavedSearchesView {
+    func configure(with savedSearches: [FrontPageSavedSearchViewModel]) {
```

Comment:
I'm just really curious as to why these extensions aren't just part of the class. It's consistent, so probably a reason for it?

Reply:
yes, as mentioned earlier it was just the coding style of the original developer of the files. Just another preference, I guess 😄  it's nice for separating concerns, so kept it as it was